src/transaction/log_replication_atomic.cpp

```diff
@@ -402,41 +407,114 @@ namespace cublog
 	assert_release (false);
       }
 
-#if !defined (NDEBUG)
-    /* there will be no duplicated object in the map */
-    auto [begin, end] = m_locked_classes.equal_range (trid);
-    if (std::count_if (begin, end, [classoid] (const auto &it)
-    {
-      return it.second == *classoid;
-    }) > 0)
-    {
-      assert (false);
-    }
-#endif
+    m_locked_classes[trid].emplace (*classoid);
+  }
+
+  void
+  atomic_replicator::bookkeep_classname (cubthread::entry &thread_entry, const OID *classoid)
+  {
+    char *classname = nullptr;
 
-    m_locked_classes.emplace (trid, *classoid);
+    assert (!OID_ISNULL (classoid) && !OID_ISTEMP (classoid));
+
+    (void) heap_get_class_name (&thread_entry, classoid, &classname);
+    if (classname != nullptr)
+      {
+	m_classname_map.emplace (*classoid, std::move (std::string (classname)));
```

Comment:
```suggestion
	m_classname_map.emplace (*classoid, std::string (classname));
```
I don't think we need to cast it to R-ref since it's already R-ref. Please correct me if I'm wrong.

Reply:
I agree, and I changed it.